Synthesis failed for nodejs-error-reporting

Hello! Autosynth couldn't regenerate nodejs-error-reporting. :broken_heart:

Please investigate and fix this issue within 5 business days.  While it remains broken,
this library cannot be updated with changes to the nodejs-error-reporting API, and the library grows
stale.

See https://github.com/googleapis/synthtool/blob/master/autosynth/TroubleShooting.md
for trouble shooting tips.

Here's the output from running `synth.py`:

```
lable at @hapi/hoek. Please update your dependencies as this version is no longer maintained an may contain bugs and security issues.
npm WARN deprecated cheerio-select-tmp@0.1.1: Use cheerio-select instead
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated opn@6.0.0: The package has been renamed to `open`
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated core-js@2.6.12: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated

> dtrace-provider@0.8.8 install /home/kbuilder/.cache/synthtool/nodejs-error-reporting/node_modules/dtrace-provider
> node-gyp rebuild || node suppress-error.js

make: Entering directory `/home/kbuilder/.cache/synthtool/nodejs-error-reporting/node_modules/dtrace-provider/build'
  TOUCH Release/obj.target/DTraceProviderStub.stamp
make: Leaving directory `/home/kbuilder/.cache/synthtool/nodejs-error-reporting/node_modules/dtrace-provider/build'

> core-js@2.6.12 postinstall /home/kbuilder/.cache/synthtool/nodejs-error-reporting/node_modules/core-js
> node -e "try{require('./postinstall')}catch(e){}"


> @compodoc/compodoc@1.1.11 postinstall /home/kbuilder/.cache/synthtool/nodejs-error-reporting/node_modules/@compodoc/compodoc
> opencollective-postinstall || exit 0

[96m[1mThank you for using @compodoc/compodoc![96m[1m
[0m[96mIf you rely on this package, please consider supporting our open collective:[22m[39m
> [94mhttps://opencollective.com/compodoc/donate[0m


> @google-cloud/error-reporting@2.0.1 prepare /home/kbuilder/.cache/synthtool/nodejs-error-reporting
> npm run compile


> @google-cloud/error-reporting@2.0.1 precompile /home/kbuilder/.cache/synthtool/nodejs-error-reporting
> gts clean

version: 14
Removing build ...

> @google-cloud/error-reporting@2.0.1 compile /home/kbuilder/.cache/synthtool/nodejs-error-reporting
> tsc -p .

[96mnode_modules/@types/hapi__hapi/index.d.ts[0m:[93m260[0m:[93m18[0m - [91merror[0m[90m TS2430: [0mInterface 'RequestEvents' incorrectly extends interface 'Podium'.
  Types of property 'on' are incompatible.
    Type '{ (criteria: "peek", listener: PeekListener): void; (criteria: "finish" | "disconnect", listener: (data: undefined) => void): void; }' is not assignable to type '<Tcontext extends object = this>(criteria: string | CriteriaObject, listener: Listener<Tcontext>, context?: Tcontext | undefined) => this'.
      Types of parameters 'listener' and 'listener' are incompatible.
        Types of parameters 'tags' and 'encoding' are incompatible.
          Type 'string' is not assignable to type '{ [tag: string]: true | undefined; } | undefined'.

[7m260[0m export interface RequestEvents extends Podium {
[7m   [0m [91m                 ~~~~~~~~~~~~~[0m

[96mnode_modules/@types/hapi__hapi/index.d.ts[0m:[93m633[0m:[93m18[0m - [91merror[0m[90m TS2430: [0mInterface 'ResponseEvents' incorrectly extends interface 'Podium'.
  Types of property 'on' are incompatible.
    Type '{ (criteria: "peek", listener: PeekListener): void; (criteria: "finish", listener: (data: undefined) => void): void; }' is not assignable to type '<Tcontext extends object = this>(criteria: string | CriteriaObject, listener: Listener<Tcontext>, context?: Tcontext | undefined) => this'.
      Types of parameters 'listener' and 'listener' are incompatible.
        Types of parameters 'tags' and 'encoding' are incompatible.
          Type 'string' is not assignable to type '{ [tag: string]: true | undefined; } | undefined'.

[7m633[0m export interface ResponseEvents extends Podium {
[7m   [0m [91m                 ~~~~~~~~~~~~~~[0m

[96mnode_modules/@types/hapi__hapi/index.d.ts[0m:[93m2363[0m:[93m43[0m - [91merror[0m[90m TS2314: [0mGeneric type 'Listener' requires 1 type argument(s).

[7m2363[0m     removeListener(criteria: K, listener: Podium.Listener): this;
[7m    [0m [91m                                          ~~~~~~~~~~~~~~~[0m

[96mnode_modules/@types/hapi__hapi/index.d.ts[0m:[93m2381[0m:[93m18[0m - [91merror[0m[90m TS2430: [0mInterface 'ServerEvents' incorrectly extends interface 'Podium'.
  The types returned by 'on(...)' are incompatible between these types.
    Type 'void' is not assignable to type 'this'.
      'this' could be instantiated with an arbitrary type which could be unrelated to 'void'.

[7m2381[0m export interface ServerEvents extends Podium {
[7m    [0m [91m                 ~~~~~~~~~~~~[0m

[96mnode_modules/@types/hapi__hapi/index.d.ts[0m:[93m2447[0m:[93m44[0m - [91merror[0m[90m TS2314: [0mGeneric type 'Listener' requires 1 type argument(s).

[7m2447[0m     removeListener(name: string, listener: Podium.Listener): Podium;
[7m    [0m [91m                                           ~~~~~~~~~~~~~~~[0m


Found 5 errors.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/error-reporting@2.0.1 compile: `tsc -p .`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/error-reporting@2.0.1 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2021-03-30T10_09_40_872Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/error-reporting@2.0.1 prepare: `npm run compile`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/error-reporting@2.0.1 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2021-03-30T10_09_40_939Z-debug.log
2021-03-30 03:09:40,969 synthtool [ERROR] > Failed executing npm install:

None
ERROR:synthtool:Failed executing npm install:

None
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/nodejs-error-reporting/synth.py", line 27, in <module>
    node.install()
  File "/tmpfs/src/github/synthtool/synthtool/languages/node.py", line 171, in install
    shell.run(["npm", "install"], hide_output=hide_output)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['npm', 'install']' returned non-zero exit status 1.
2021-03-30 03:09:41,042 autosynth [ERROR] > Synthesis failed
2021-03-30 03:09:41,043 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at 419a5e1 test: fix Mocha patch with undefined timeout error (#547)
2021-03-30 03:09:41,051 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2021-03-30 03:09:41,057 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Removing node_modules/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 354, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 189, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 334, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 65, in synthesize_loop
    has_changes = toolbox.synthesize_version_in_new_branch(synthesizer, youngest)
  File "/tmpfs/src/github/synthtool/autosynth/synth_toolbox.py", line 259, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/2562fe77-7557-4a05-b10c-ac5fc1994233/targets/github%2Fsynthtool;config=default/tests;query=nodejs-error-reporting;failed=false).

Autosynth is still having trouble generating nodejs-error-reporting. :sob:

Please investigate and fix this issue within 5 business days.  While it remains broken,
this library cannot be updated with changes to the nodejs-error-reporting API, and the library grows
stale.

See https://github.com/googleapis/synthtool/blob/master/autosynth/TroubleShooting.md
for trouble shooting tips.

Here's the output from running `synth.py`:

```
lable at @hapi/hoek. Please update your dependencies as this version is no longer maintained an may contain bugs and security issues.
npm WARN deprecated cheerio-select-tmp@0.1.1: Use cheerio-select instead
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated opn@6.0.0: The package has been renamed to `open`
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated core-js@2.6.12: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated

> dtrace-provider@0.8.8 install /home/kbuilder/.cache/synthtool/nodejs-error-reporting/node_modules/dtrace-provider
> node-gyp rebuild || node suppress-error.js

make: Entering directory `/home/kbuilder/.cache/synthtool/nodejs-error-reporting/node_modules/dtrace-provider/build'
  TOUCH Release/obj.target/DTraceProviderStub.stamp
make: Leaving directory `/home/kbuilder/.cache/synthtool/nodejs-error-reporting/node_modules/dtrace-provider/build'

> core-js@2.6.12 postinstall /home/kbuilder/.cache/synthtool/nodejs-error-reporting/node_modules/core-js
> node -e "try{require('./postinstall')}catch(e){}"


> @compodoc/compodoc@1.1.11 postinstall /home/kbuilder/.cache/synthtool/nodejs-error-reporting/node_modules/@compodoc/compodoc
> opencollective-postinstall || exit 0

[96m[1mThank you for using @compodoc/compodoc![96m[1m
[0m[96mIf you rely on this package, please consider supporting our open collective:[22m[39m
> [94mhttps://opencollective.com/compodoc/donate[0m


> @google-cloud/error-reporting@2.0.1 prepare /home/kbuilder/.cache/synthtool/nodejs-error-reporting
> npm run compile


> @google-cloud/error-reporting@2.0.1 precompile /home/kbuilder/.cache/synthtool/nodejs-error-reporting
> gts clean

version: 14
Removing build ...

> @google-cloud/error-reporting@2.0.1 compile /home/kbuilder/.cache/synthtool/nodejs-error-reporting
> tsc -p .

[96mnode_modules/@types/hapi__hapi/index.d.ts[0m:[93m260[0m:[93m18[0m - [91merror[0m[90m TS2430: [0mInterface 'RequestEvents' incorrectly extends interface 'Podium'.
  Types of property 'on' are incompatible.
    Type '{ (criteria: "peek", listener: PeekListener): void; (criteria: "finish" | "disconnect", listener: (data: undefined) => void): void; }' is not assignable to type '<Tcontext extends object = this>(criteria: string | CriteriaObject, listener: Listener<Tcontext>, context?: Tcontext | undefined) => this'.
      Types of parameters 'listener' and 'listener' are incompatible.
        Types of parameters 'tags' and 'encoding' are incompatible.
          Type 'string' is not assignable to type '{ [tag: string]: true | undefined; } | undefined'.

[7m260[0m export interface RequestEvents extends Podium {
[7m   [0m [91m                 ~~~~~~~~~~~~~[0m

[96mnode_modules/@types/hapi__hapi/index.d.ts[0m:[93m633[0m:[93m18[0m - [91merror[0m[90m TS2430: [0mInterface 'ResponseEvents' incorrectly extends interface 'Podium'.
  Types of property 'on' are incompatible.
    Type '{ (criteria: "peek", listener: PeekListener): void; (criteria: "finish", listener: (data: undefined) => void): void; }' is not assignable to type '<Tcontext extends object = this>(criteria: string | CriteriaObject, listener: Listener<Tcontext>, context?: Tcontext | undefined) => this'.
      Types of parameters 'listener' and 'listener' are incompatible.
        Types of parameters 'tags' and 'encoding' are incompatible.
          Type 'string' is not assignable to type '{ [tag: string]: true | undefined; } | undefined'.

[7m633[0m export interface ResponseEvents extends Podium {
[7m   [0m [91m                 ~~~~~~~~~~~~~~[0m

[96mnode_modules/@types/hapi__hapi/index.d.ts[0m:[93m2363[0m:[93m43[0m - [91merror[0m[90m TS2314: [0mGeneric type 'Listener' requires 1 type argument(s).

[7m2363[0m     removeListener(criteria: K, listener: Podium.Listener): this;
[7m    [0m [91m                                          ~~~~~~~~~~~~~~~[0m

[96mnode_modules/@types/hapi__hapi/index.d.ts[0m:[93m2381[0m:[93m18[0m - [91merror[0m[90m TS2430: [0mInterface 'ServerEvents' incorrectly extends interface 'Podium'.
  The types returned by 'on(...)' are incompatible between these types.
    Type 'void' is not assignable to type 'this'.
      'this' could be instantiated with an arbitrary type which could be unrelated to 'void'.

[7m2381[0m export interface ServerEvents extends Podium {
[7m    [0m [91m                 ~~~~~~~~~~~~[0m

[96mnode_modules/@types/hapi__hapi/index.d.ts[0m:[93m2447[0m:[93m44[0m - [91merror[0m[90m TS2314: [0mGeneric type 'Listener' requires 1 type argument(s).

[7m2447[0m     removeListener(name: string, listener: Podium.Listener): Podium;
[7m    [0m [91m                                           ~~~~~~~~~~~~~~~[0m


Found 5 errors.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/error-reporting@2.0.1 compile: `tsc -p .`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/error-reporting@2.0.1 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2021-03-31T10_10_15_571Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/error-reporting@2.0.1 prepare: `npm run compile`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/error-reporting@2.0.1 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2021-03-31T10_10_15_623Z-debug.log
2021-03-31 03:10:15,664 synthtool [ERROR] > Failed executing npm install:

None
ERROR:synthtool:Failed executing npm install:

None
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/nodejs-error-reporting/synth.py", line 27, in <module>
    node.install()
  File "/tmpfs/src/github/synthtool/synthtool/languages/node.py", line 171, in install
    shell.run(["npm", "install"], hide_output=hide_output)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['npm', 'install']' returned non-zero exit status 1.
2021-03-31 03:10:15,708 autosynth [ERROR] > Synthesis failed
2021-03-31 03:10:15,709 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at 419a5e1 test: fix Mocha patch with undefined timeout error (#547)
2021-03-31 03:10:15,716 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2021-03-31 03:10:15,722 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Removing node_modules/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 354, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 189, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 334, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 65, in synthesize_loop
    has_changes = toolbox.synthesize_version_in_new_branch(synthesizer, youngest)
  File "/tmpfs/src/github/synthtool/autosynth/synth_toolbox.py", line 259, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/de9126dd-84ba-4c32-8803-c5bc0265729b/targets/github%2Fsynthtool;config=default/tests;query=nodejs-error-reporting;failed=false).

Autosynth is still having trouble generating nodejs-error-reporting. :sob:

Please investigate and fix this issue within 5 business days.  While it remains broken,
this library cannot be updated with changes to the nodejs-error-reporting API, and the library grows
stale.

See https://github.com/googleapis/synthtool/blob/master/autosynth/TroubleShooting.md
for trouble shooting tips.

Here's the output from running `synth.py`:

```
lable at @hapi/hoek. Please update your dependencies as this version is no longer maintained an may contain bugs and security issues.
npm WARN deprecated cheerio-select-tmp@0.1.1: Use cheerio-select instead
npm WARN deprecated chokidar@2.1.8: Chokidar 2 will break on node v14+. Upgrade to chokidar 3 with 15x less dependencies.
npm WARN deprecated opn@6.0.0: The package has been renamed to `open`
npm WARN deprecated fsevents@1.2.13: fsevents 1 will break on node v14+ and could be using insecure binaries. Upgrade to fsevents 2.
npm WARN deprecated core-js@2.6.12: core-js@<3 is no longer maintained and not recommended for usage due to the number of issues. Please, upgrade your dependencies to the actual version of core-js@3.
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated

> dtrace-provider@0.8.8 install /home/kbuilder/.cache/synthtool/nodejs-error-reporting/node_modules/dtrace-provider
> node-gyp rebuild || node suppress-error.js

make: Entering directory `/home/kbuilder/.cache/synthtool/nodejs-error-reporting/node_modules/dtrace-provider/build'
  TOUCH Release/obj.target/DTraceProviderStub.stamp
make: Leaving directory `/home/kbuilder/.cache/synthtool/nodejs-error-reporting/node_modules/dtrace-provider/build'

> core-js@2.6.12 postinstall /home/kbuilder/.cache/synthtool/nodejs-error-reporting/node_modules/core-js
> node -e "try{require('./postinstall')}catch(e){}"


> @compodoc/compodoc@1.1.11 postinstall /home/kbuilder/.cache/synthtool/nodejs-error-reporting/node_modules/@compodoc/compodoc
> opencollective-postinstall || exit 0

[96m[1mThank you for using @compodoc/compodoc![96m[1m
[0m[96mIf you rely on this package, please consider supporting our open collective:[22m[39m
> [94mhttps://opencollective.com/compodoc/donate[0m


> @google-cloud/error-reporting@2.0.1 prepare /home/kbuilder/.cache/synthtool/nodejs-error-reporting
> npm run compile


> @google-cloud/error-reporting@2.0.1 precompile /home/kbuilder/.cache/synthtool/nodejs-error-reporting
> gts clean

version: 14
Removing build ...

> @google-cloud/error-reporting@2.0.1 compile /home/kbuilder/.cache/synthtool/nodejs-error-reporting
> tsc -p .

[96mnode_modules/@types/hapi__hapi/index.d.ts[0m:[93m260[0m:[93m18[0m - [91merror[0m[90m TS2430: [0mInterface 'RequestEvents' incorrectly extends interface 'Podium'.
  Types of property 'on' are incompatible.
    Type '{ (criteria: "peek", listener: PeekListener): void; (criteria: "finish" | "disconnect", listener: (data: undefined) => void): void; }' is not assignable to type '<Tcontext extends object = this>(criteria: string | CriteriaObject, listener: Listener<Tcontext>, context?: Tcontext | undefined) => this'.
      Types of parameters 'listener' and 'listener' are incompatible.
        Types of parameters 'tags' and 'encoding' are incompatible.
          Type 'string' is not assignable to type '{ [tag: string]: true | undefined; } | undefined'.

[7m260[0m export interface RequestEvents extends Podium {
[7m   [0m [91m                 ~~~~~~~~~~~~~[0m

[96mnode_modules/@types/hapi__hapi/index.d.ts[0m:[93m633[0m:[93m18[0m - [91merror[0m[90m TS2430: [0mInterface 'ResponseEvents' incorrectly extends interface 'Podium'.
  Types of property 'on' are incompatible.
    Type '{ (criteria: "peek", listener: PeekListener): void; (criteria: "finish", listener: (data: undefined) => void): void; }' is not assignable to type '<Tcontext extends object = this>(criteria: string | CriteriaObject, listener: Listener<Tcontext>, context?: Tcontext | undefined) => this'.
      Types of parameters 'listener' and 'listener' are incompatible.
        Types of parameters 'tags' and 'encoding' are incompatible.
          Type 'string' is not assignable to type '{ [tag: string]: true | undefined; } | undefined'.

[7m633[0m export interface ResponseEvents extends Podium {
[7m   [0m [91m                 ~~~~~~~~~~~~~~[0m

[96mnode_modules/@types/hapi__hapi/index.d.ts[0m:[93m2363[0m:[93m43[0m - [91merror[0m[90m TS2314: [0mGeneric type 'Listener' requires 1 type argument(s).

[7m2363[0m     removeListener(criteria: K, listener: Podium.Listener): this;
[7m    [0m [91m                                          ~~~~~~~~~~~~~~~[0m

[96mnode_modules/@types/hapi__hapi/index.d.ts[0m:[93m2381[0m:[93m18[0m - [91merror[0m[90m TS2430: [0mInterface 'ServerEvents' incorrectly extends interface 'Podium'.
  The types returned by 'on(...)' are incompatible between these types.
    Type 'void' is not assignable to type 'this'.
      'this' could be instantiated with an arbitrary type which could be unrelated to 'void'.

[7m2381[0m export interface ServerEvents extends Podium {
[7m    [0m [91m                 ~~~~~~~~~~~~[0m

[96mnode_modules/@types/hapi__hapi/index.d.ts[0m:[93m2447[0m:[93m44[0m - [91merror[0m[90m TS2314: [0mGeneric type 'Listener' requires 1 type argument(s).

[7m2447[0m     removeListener(name: string, listener: Podium.Listener): Podium;
[7m    [0m [91m                                           ~~~~~~~~~~~~~~~[0m


Found 5 errors.

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/error-reporting@2.0.1 compile: `tsc -p .`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/error-reporting@2.0.1 compile script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2021-04-01T13_02_43_366Z-debug.log
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! @google-cloud/error-reporting@2.0.1 prepare: `npm run compile`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the @google-cloud/error-reporting@2.0.1 prepare script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/kbuilder/.npm/_logs/2021-04-01T13_02_43_414Z-debug.log
2021-04-01 06:02:43,452 synthtool [ERROR] > Failed executing npm install:

None
ERROR:synthtool:Failed executing npm install:

None
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 102, in <module>
    main()
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 829, in __call__
    return self.main(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 782, in main
    rv = self.invoke(ctx)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 1066, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/tmpfs/src/github/synthtool/env/lib/python3.6/site-packages/click/core.py", line 610, in invoke
    return callback(*args, **kwargs)
  File "/tmpfs/src/github/synthtool/synthtool/__main__.py", line 94, in main
    spec.loader.exec_module(synth_module)  # type: ignore
  File "<frozen importlib._bootstrap_external>", line 678, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/kbuilder/.cache/synthtool/nodejs-error-reporting/synth.py", line 27, in <module>
    node.install()
  File "/tmpfs/src/github/synthtool/synthtool/languages/node.py", line 171, in install
    shell.run(["npm", "install"], hide_output=hide_output)
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 39, in run
    raise exc
  File "/tmpfs/src/github/synthtool/synthtool/shell.py", line 33, in run
    encoding="utf-8",
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 438, in run
    output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['npm', 'install']' returned non-zero exit status 1.
2021-04-01 06:02:43,494 autosynth [ERROR] > Synthesis failed
2021-04-01 06:02:43,494 autosynth [DEBUG] > Running: git reset --hard HEAD
HEAD is now at 419a5e1 test: fix Mocha patch with undefined timeout error (#547)
2021-04-01 06:02:43,502 autosynth [DEBUG] > Running: git checkout autosynth
Switched to branch 'autosynth'
2021-04-01 06:02:43,507 autosynth [DEBUG] > Running: git clean -fdx
Removing __pycache__/
Removing node_modules/
Traceback (most recent call last):
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 193, in _run_module_as_main
    "__main__", mod_spec)
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/runpy.py", line 85, in _run_code
    exec(code, run_globals)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 356, in <module>
    main()
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 191, in main
    return _inner_main(temp_dir)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 336, in _inner_main
    commit_count = synthesize_loop(x, multiple_prs, change_pusher, synthesizer)
  File "/tmpfs/src/github/synthtool/autosynth/synth.py", line 68, in synthesize_loop
    has_changes = toolbox.synthesize_version_in_new_branch(synthesizer, youngest)
  File "/tmpfs/src/github/synthtool/autosynth/synth_toolbox.py", line 259, in synthesize_version_in_new_branch
    synthesizer.synthesize(synth_log_path, self.environ)
  File "/tmpfs/src/github/synthtool/autosynth/synthesizer.py", line 120, in synthesize
    synth_proc.check_returncode()  # Raise an exception.
  File "/home/kbuilder/.pyenv/versions/3.6.9/lib/python3.6/subprocess.py", line 389, in check_returncode
    self.stderr)
subprocess.CalledProcessError: Command '['/tmpfs/src/github/synthtool/env/bin/python3', '-m', 'synthtool', '--metadata', 'synth.metadata', 'synth.py', '--']' returned non-zero exit status 1.

```

Google internal developers can see the full log [here](http://sponge2/results/invocations/edf58730-9e12-4d2c-b179-a519d36eadb8/targets/github%2Fsynthtool;config=default/tests;query=nodejs-error-reporting;failed=false).

--- a/tsconfig.json
+++ b/tsconfig.json
@@ -3,7 +3,13 @@
   "compilerOptions": {
     "lib": ["es2018", "dom"],
     "rootDir": ".",
-    "outDir": "build"
+    "outDir": "build",
+    "baseUrl": ".",
+    "paths": {
+      "@hapi/podium": [
+        "node_modules/@types/hapi__podium"
+      ]
+    }
   },
   "include": [
     "src/*.ts",

